sstables/sstable_directory.cc

```diff
@@ -558,29 +558,18 @@ bool sstable_directory::compare_sstable_storage_prefix(const sstring& prefix_a,
     return size_a == size_b && sstring::traits_type::compare(prefix_a.begin(), prefix_b.begin(), size_a) == 0;
 }
 
-future<std::pair<sstring, sstring>> sstable_directory::create_pending_deletion_log(const std::vector<shared_sstable>& ssts) {
+future<std::unordered_map<sstring, sstring>> sstable_directory::create_pending_deletion_log(const std::vector<shared_sstable>& ssts) {
     return seastar::async([&ssts] {
-        shared_sstable first = nullptr;
-        min_max_tracker<generation_type> gen_tracker;
+        std::unordered_map<sstring, min_max_tracker<generation_type>> gen_trackers;
+        std::unordered_map<sstring, sstring> res;
 
         for (const auto& sst : ssts) {
-            gen_tracker.update(sst->generation());
-
-            if (first == nullptr) {
-                first = sst;
-            } else {
-                // All sstables are assumed to be in the same column_family, hence
-                // sharing their base directory. Since lexicographical comparison of
-                // paths is not the same as their actually equivalence, this should
-                // rather check for fs::equivalent call on _storage.prefix()-s. But
-                // since we know that the worst thing filesystem storage driver can
-                // do is to prepend/drop the trailing slash, it should be enough to
-                // compare prefixes of both ... prefixes
-                assert(compare_sstable_storage_prefix(first->_storage->prefix(), sst->_storage->prefix()));
-            }
+            auto prefix = sst->_storage->prefix();
+            gen_trackers[prefix].update(sst->generation());
         }
 
-        sstring pending_delete_dir = first->_storage->prefix() + "/" + sstables::pending_delete_dir;
+      for (const auto& [prefix, gen_tracker] : gen_trackers) {
+        sstring pending_delete_dir = prefix + "/" + sstables::pending_delete_dir;
         sstring pending_delete_log = format("{}/sstables-{}-{}.log", pending_delete_dir, gen_tracker.min(), gen_tracker.max());
```

Comment:
Just curious, did we update the log scanner for uuid generations?

Reply:
We currently don't really care about the generations used in the pending delete log name when replaying it.
They're used for uniqueness and potentially if we find good use for them in the future.